src/NLog/Internal/XmlHelper.cs

```diff
@@ -99,6 +99,99 @@ private static string CreateValidXmlString(string text)
 #endif
 
         /// <summary>
+        /// Converts object value to invariant format, and strips any invalid xml-characters
+        /// </summary>
+        /// <param name="value">Object value</param>
+        /// <returns>Object value converted to string</returns>
+        internal static string XmlConvertToStringSafe(object value)
+        {
+            string valueString = XmlConvertToString(value);
+            return RemoveInvalidXmlChars(valueString);
+        }
+
+        /// <summary>
+        /// Converts object value to invariant format (understood by JavaScript)
+        /// </summary>
+        /// <param name="value">Object value</param>
+        /// <returns>Object value converted to string</returns>
+        internal static string XmlConvertToString(object value)
+        {
+            TypeCode objTypeCode;
+            return XmlConvertToString(value, out objTypeCode);
+        }
+
+        /// <summary>
+        /// Converts object value to invariant format (understood by JavaScript)
+        /// </summary>
+        /// <param name="value">Object value</param>
+        /// <param name="objTypeCode">Object TypeCode</param>
+        /// <returns>Object value converted to string</returns>
+        internal static string XmlConvertToString(object value, out TypeCode objTypeCode)
+        {
+            if (value == null)
+            {
+                objTypeCode = TypeCode.Empty;
+                return "null";
+            }
+
+            objTypeCode = Convert.GetTypeCode(value);
+            switch (objTypeCode)
```

Comment:
Think switch-statement is faster than dictionary-lookup. System.Convert.GetTypeCode should also work in NET Core (It is System.Type.GetTypeCode that has been removed).